### ISSUE TITLE
[TextInputLayout] Graphical issues with rounded TextInputLayout hint.

### ISSUE THREAD

**Issue by beneventolorenzo** [material-components/material-components-android#2145]
**Description:** The inset for the label is misplaced and doesn't fully cover the stroke of the container on the Outlined Text Field if corners are rounded. 

**Expected behavior:** The inset should cover the stroke.

**Source code:** No need for code in my opinion, but I'm absolutely available to provide some code if needed.

**Android API version:** R

**Material Library version:** Was fine on 1.3.0, broken on 1.4.0-alpha01

**Device:** Both emulator and physical One Plus Nord.
![photo_2021-03-22_13-32-20](https://user-images.githubusercontent.com/6758238/111990241-73e59f80-8b13-11eb-8913-90fc95239c1d.jpg)

**Comment by drchen** [comment]
It's introduced by https://github.com/material-components/material-components-android/pull/1660

Not sure if there's a way to serve both situations. Investigating.

**Comment by beneventolorenzo** [comment]
Makes sense, it's still there in 1.4.0 of course. A workaround may be to manually set the background of the box to match the background color then, right?

**Comment by drchen** [comment]
Not sure if there's a good workaround with the current version, except for adjusting corner size so the Android system won't draw "beyond the bound". 

One idea is to provide an extra flag to turn on/off this "exact" cutout behavior but we need some discussion among our team.

**Comment by beneventolorenzo** [comment]
Yes, I realized maybe there is not. An extra flag to turn on/off this behaviour would be great. Another possible approach would be to have an attribute to set a colour on the box surrounding the text! I'm gonna stick with 1.3.0 for now. I'm supposing any news about this issue will be posted here so I'm gonna make sure I keep an eye on it. If you need something I'll be happy to help!
